datadog/threadstats/metrics.py

```diff
@@ -139,7 +139,7 @@ def add_point(self, metric, tags, timestamp, value, metric_class, sample_rate=1,
     def flush(self, timestamp):
         """ Flush all metrics up to the given timestamp. """
         interval = timestamp - timestamp % self._roll_up_interval
-        past_intervals = [i for i in self._metrics if i < interval]
+        past_intervals = [i for i in list(self._metrics.keys()) if i < interval]
```

Comment:
`self._metrics.keys()` already creates a copy of `self._metrics` keys. Let's remove the extra `list(...)`.